playbooks/roles/logstash/templates/logstash.conf.j2

```diff
@@ -1,43 +1,196 @@
+    grok {
+      match => { "message" => "%{WORD}\=%{QUOTEDSTRING:syslog_timestamp} %{WORD}\=%{QUOTEDSTRING:rabbitmq_vhost} %{WORD}\=%{QUOTEDSTRING:rabbitmq_queue} %{WORD}\=%{INT:rabbitmq_length}" }
+    }
+  } else if ("{{ COMMON_LOG_DIR }}/mongo/mongodb.log" in [source]) {
+    grok {
+      match => { "message" => "%{TIMESTAMP_ISO8601:syslog_timestamp} ?\[?%{DATA:moongo_context}?\]? %{GREEDYDATA:syslog_message}" }
+    } 
+  } else if ("{{ COMMON_LOG_DIR }}/tracking/tracking.log" in [source] or 
+    "/edx/app/edxapp/edx-platform/logs/app1/tracking/tracking.log" in [source]) {
+    grok {
+      match => { "message" => "%{GREEDYDATA:syslog_message}" }
+    }
+  } else if ("{{ logstash_log_dir }}/logstash.log" in [source]) {
+    grok {
+      match => { "message" => "\{\:%{WORD}\=\>\"%{TIMESTAMP_ISO8601:syslog_timestamp}\"\, %{GREEDYDATA:syslog_message}\}" }
+    }
+  } else if ("{{ elasticsearch_log_dir }}/elasticsearch.log" in [source]) {
+    grok {
+      match => { "message" => "(?:\[%{TIMESTAMP_ISO8601:syslog_timestamp}\]\[%{WORD:log_level}? \])%{GREEDYDATA:syslog_message}" }
+    }
+  } else { #matches all rsyslog messages
+    grok {
+      match => { "message" => "%{SYSLOGTIMESTAMP:syslog_timestamp} %{WORD:syslog_hostname} %{DATA:syslog_program}(?:\[%{POSINT:syslog_pid}\])?: %{GREEDYDATA:syslog_message}" }
+      match => { "message" => "%{SYSLOGTIMESTAMP:syslog_timestamp} %{WORD:syslog_hostname} \[%{WORD}\=%{DATA:service_variant}\]\[%{DATA:service_program}\]\[%{WORD}\:%{USERNAME:env}\] %{WORD:log_level} \[%{WORD} ? %{POSINT:syslog_pid}\] \[%{DATA:filename}\:%{INT:line_number}\] \- %{GREEDYDATA:syslog_message}" }
+      tag_on_failure => []
+    }
+  }
+  }
+  syslog_pri { }
+  date {
+    match => [ "syslog_timestamp", "MMM  d HH:mm:ss", "MMM dd HH:mm:ss", "DD MMM YYYY:HH:mm:ss ZZZ", "ISO8601"]
+  }
+  # Try and parse the tracking log json
+  # 142 is syslog facility 17 (local1) and Informational.
+  # This is used to reduce the number of errors in json parsing as
+  # tracking uses that facility and priority by default.
+  if "142" in [syslog_pri] {
+    json {
+      source => "syslog_message"
+      target => "tracking"
+    }
+  }
+#  if ([message] =~ /(?i)warn/) {
+#    mutate {
+#      add_tag => "warning"
+#    }
+#  }
+#  if ([message] =~ /(?i)error/) {
+#    mutate {
+#      add_tag => " error"
+#    }
+#  }
+  if ([message] =~ /(?i)forums/) {
+    mutate {
+      add_tag => "forums"
+    }
+  }
+  if ([message] =~ /(?i)certs/) {
+    mutate {
+      add_tag => "certs"
+    }
+  }
+  if ("cms" in [service_variant]) {
+    mutate {
+      add_tag => "cms"
+    }
+  }
+  if ("lms" in [service_variant]) {
+    mutate {
+      add_tag => "lms"
+    }
+  }
+  if !("_grokparsefailure" in [tags]) {
+    mutate {
+      replace => [ "@source_host", "%{host}" ]
     }
     mutate {
-      remove_field => [ "syslog_hostname", "syslog_message", "syslog_timestamp" ]
```

Comment:
Now we write straight to @message, instead of writing to syslog_message copying into @message, and then deleting syslog_message. I changed this since the vast majority of our logs do not have syslog_messages so it made sense to use the more standard term.